src/read.c

```diff
@@ -2573,11 +2565,7 @@ ExecStatus ReadEvalCommand(Obj context, Obj *evalResult, UInt *dualSemicolon)
 
     /* switch back to the old reader context                               */
     memcpy( STATE(ReadJmpError), readJmpError, sizeof(syJmp_buf) );
-    rs->StackNams      = stackNams;
-    rs->ReadTop        = readTop;
-    rs->ReadTilde      = readTilde;
     STATE(Tilde)       = tilde;
-    rs->CurrLHSGVar    = currLHSGVar;
```

Comment:
Real change: since we freshly allocated the `ReaderState` on the stack, there is no need to "restore" its content before returning.